elements/pfe-primary-detail/pfe-primary-detail.ts

```diff
@@ -0,0 +1,667 @@
+  consistentHeight = false;
+
+  /** The min-width of the _component_ where it has a desktop layout */
+  @property({ attribute: 'breakpoint-width', type: Number, reflect: true }) breakpointWidth = 800;
+
+  /** Read only: Displays the id of an open 'detailNav' element */
+  @observed
+  @property({ type: String, reflect: true }) active: string|null = null;
+
+  /** Read only: Displays what breakpoint is currently being used */
+  @property({ type: String, reflect: true }) breakpoint?: 'compact'|'desktop';
+
+  @state() private pristine = true;
+  @state() private controlsId?: string;
+  @state() private nextToggle: HTMLElement|null = null;
+
+  @query('#details-wrapper__back') private _detailsBackButton?: HTMLElement|null;
+
+  private logger = new Logger(this);
+
+  private get root(): Document|ShadowRoot|null {
+    const root = this.getRootNode();
+    if (!isQueryable(root))
+      return null;
+    else
+      return root;
+  }
+
+  private slots = new SlotController(this, [
+    // Content that shows above the navigation column
+    'header',
+    // Column of headings to expland the related contents
+    'nav',
+    // Content that shows below the navigation column
+    'footer',
+    // Content content that is shown when it's corresponding "Details Nav" item is active
+    'details',
+    // deprecated
+    'details-nav--header',
+    'details-nav--footer',
+    'details-nav',
+  ]);
+
+  private _debouncedSetBreakpoint:ReturnType<typeof debounce>|null = null;
+
+  private _windowInnerWidth?: number;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.setAttribute('aria-orientation', 'vertical');
+    this.setAttribute('role', 'tablist');
+
+    // Lower debounce delay for automated testing
+    const debounceDelay = this.hasAttribute('automated-testing') ? 0 : 100;
+
+    this._debouncedSetBreakpoint = debounce(this._setBreakpoint, debounceDelay);
+
+    window.addEventListener('resize', this._debouncedSetBreakpoint);
+
+    // A11y: add keydown event listener to activate keyboard controls
+    this.addEventListener('keydown', this._keyboardControls);
+  }
+
+  render() {
+    return html`
+      <div id="wrapper"
+          class="${classMap({ active: !!this.active })}"
+          ?data-pristine="${this.pristine}">
+        <div id="details-nav">
+          <slot name="header"></slot>
+          <slot name="details-nav--header"></slot>
+          <slot name="nav"></slot>
+          <slot name="details-nav"></slot>
+        </div>
+        <div id="details-wrapper">
+          <div id="details-wrapper__header">
+            <button
+                id="details-wrapper__back"
+                aria-controls="${ifDefined(this.controlsId)}"
+                aria-expanded="${ifDefined(this.breakpoint !== 'compact' ? undefined : String(this.active))}"
+                @click="${this.closeAll}">
+              ${this._renderDetailsWrapperHeading()}
```

Comment:
There seems to be a race condition where this isn't being set before render... not sure how to debug that?

It's being set on this line in `_handleHideShow` on line 457 of this file:
https://github.com/patternfly/patternfly-elements/pull/1804/commits/0f77651dbfaf06298f13461be8d876f2c8f82748#diff-a9cda0cfdb7c551c3c59f90512491b5c978994191e89f33a9ff0ef8528bad640R457

Reply:
@wesruv can you check this again against the latest (rebased) changes, please?

Reply:
Can you also check the errors thrown in the tests?

```
npm run test -- --group default --watch --files elements/pfe-primary-detail/test/primary-detail.spec.ts
```
then hit `d` to launch debug the tests in a browser